### ISSUE TITLE
SMS sent via desktop apps such as KDEConnect send but lack message history

### ISSUE THREAD

**Issue by Ace-Cassidy** [signalapp/Signal-Android#9442]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Messages sent through GSConnect or KDEConnect lack message history in Signal.
There are currently no open issues and THIS IS A BUG because this same issue effects interoperability of signal with many other services/apps such as Tasker and SMS Code Reader where one would expect successful integration with an SMS application, but instead unreliable and incomplete functionality is present.

gsconnect issue: https://github.com/andyholmes/gnome-shell-extension-gsconnect/issues/320
kdeconnect issue: https://bugs.kde.org/show_bug.cgi?id=401677
signal issues: #5491 #5729 #464 #482 #6089 #7403

### Steps to reproduce
- Send an SMS via gsconnect or kdeconnect
- View messages on senders Signal app and receivers Signal app

**Actual result:** SMS appears in receivers Signal but history does not appear in senders Signal app
**Expected result:** SMS should be sent and Signal should show sent history OR SMS should be rejected entirely

### Device info

**Device:** Google Pixel 3a
**Android version:** 10
**Signal version:** 4.55.8

**Comment by guimspace** [comment]
> There is no way for us to merge a local database view with a system database view, so we either have to store all messages (SMS or otherwise) in the system SMS database or store all messages (SMS or otherwise) in our own database. - [ moxie0 commented on Nov 3, 2017](https://github.com/signalapp/Signal-Android/issues/5491#issuecomment-341802491)

So even if gsconnect or kdeconnect put the SMS in the system database, Signal simply doesn't check for new SMSs in the system database because, well, it is the default SMS app.

**Comment by Genis-Sage** [comment]
hey @Ace-Cassidy 
even if you write bug in big letters, that does not change it

Signal never stated, that it works the way you want it work
so you want it to work in a different way, as it is designed

and this is a feature request, which should be discussed on the forum https://community.signalusers.org/

**Comment by Ace-Cassidy** [comment]
Just because the developers don't want to resolve this, assumedly because it's hard to fix, doesn't make it NOT a bug. 

There are at least 8 other related issues. Signals lack of ability to integrate with other services leads to unexpected behavior, and prevents users from adopting signal. If that's not a bug I don't know what is.

This same issue also plagues signal desktop, where sms sent to non signal contacts doesn't show up at all, even when sent through signal.

Or you can choose to ignore your users and the ever growing stack of issues related to this by arbitrarily closing them without proper discussion.

**Comment by Tursko** [comment]
The integration also applies to Google's assistant. If you ask the assistant to SMS someone or reply using assistant the message is sent via the Android SMS service but it's not stored on the thread.

**Comment by CristianKerr** [comment]
sad, I was looking at this today and it looked really great until I found out that using sms in signal doesn't integrate with... well anything... All of the sudden I do not get sms with my GSconnect (on desktop) and I can see no history. And it is not like the Signal would then show me the sms in the desktop app.. No.. And then I find bunch of bug reports that are closed with "issue cleanups" or "not a bug". 
I finally got my friends on signal and now when I start searching for issues, I find out that nobody really cares about what users want, they do it the way they want and do not listen. I am very disappointed right now and hope the attitude will change. Otherwise I will withdraw my donations and myself and go through the hell of moving my friends to something better again...
